### PR TITLE
Fix: update archive package text

### LINKED ISSUE #254: Defining a process for archiving / sunsetting pyos packages
In the astropy document related to our partnership, a question/ discussion came up as follows:

> QUESTION - what happens if the package becomes maintained again - like they are out sick for 6 months, etc. the maintainer needs to ping us to ask if they can become a vetted package again (we will need a process for this documented)

i am going to copy the comments and open for discussion here as requested by @pllim . (so sorry i missed that comment in july somehow and just saw it today. 

## Length of time until we begin to wonder about a package being maintained or not?

> How long would the "graceful sunset" period last? Some packages are small and stable and "nothing going on for 3 month" might not be a bad sign at all. Numpy is pretty good in backwards compatibility, so if nothing breaks, there might be no activity. If the sunset period is long enough (say 18 month = one major Python release cycle) then this is not going to come up often in practice.
> I image that pyopensci would reach out in the sunset period before removing the listing and say "hey, we noticed that nothing is going on, is this still maintained?". If there is no answer, that's a bad sign, but the answer might be "It's stable, no new features are needed and no new Python version has be release that could have broken it". In that case, I would argue that it should not be sunset in the first place

*  [pyos policy is here](https://www.pyopensci.org/software-peer-review/our-process/policies.html#maintainer-responsiveness) 

> My process has been so far - 
> 
> 1. i watch the commit dates on the repo to note if there is any activity.
> 2. I email after a year if i see no activity

@hamogu i'm continuing the conversation here. i've also discussed this a bit with @cmarmo - we talked about using language such as "archive" which is what @rOpenSci uses. But doing that - it feels a bit less "permanent" as in - if someone wants to pick the package back up, we could easily unarchive it. 

we also talked about

* adding an archived label to the review issue
* listing archived projects on a separate page on the website so the work that went into the project is still recognized but we don't suggest that scientists adopted unmaintained tools if they don't have to! 

Let me know what your thoughts are here so we can refine our policy!!

### PR DESCRIPTION
**Pull request by lwasser**
closes #254 

this PR addresses the topic discussed in #254 it adds language around what we track and how we flag packages that may be en route to becoming orphans / unmaintained. 

it is now open for formal review from the community. 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205816362403674
  - https://app.asana.com/0/0/1206105000277660

### CODE REVIEW

File: our-process/policies.md
Comment (hamogu) [inline our-process/policies.md:27]: I know this is just a formatting change, but should it say 
```suggestion
- publishing on `PyPI` (and/or `conda`);
```

File: our-process/policies.md
Comment (hamogu) [inline our-process/policies.md:94]: What is an "update"? Any change the repro? A tagged release?

If "update" just means "any activity in the repro e.g. discussion on some issue", I think it's fine. If "update" means "release a tagged version" 6 months is too short. Python itself releases major versions only every 18 months and numpy changes are typically small; 6 month is just one numpy release. So, a stable package without significant bugs might not need an "update" that often. I suggest 12 months.

File: our-process/policies.md
Comment (willingc) [reply our-process/policies.md:27]: Do you anticipate people to do this on every publish of a release to PyPI or `conda-forge`? Major releases or just once?

Comment (willingc) [reply our-process/policies.md:27]: Perhaps:
- publishing to a package repository, such as `PyPI` and `conda-forge`

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:85]:
```suggestion
We will reach out to our package maintainers each year to verify the package is actively maintained
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:86]:
```suggestion
and to see if there are any updates we can highlight through our social channels.
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:91]:
```suggestion
the last release and last commit to the package repository. Based on activity in
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:94]: - Packages with low activity (no new commits or releases) in the past six months will be noted by a small icon flag on the website.

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:95]:
```suggestion
- Packages with low activity (no new commits or releases) for the past 12 months will be flagged for potential archive.
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:96]:
```suggestion
  A pyOpenSci team member will contact the
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:104]:
```suggestion
the package's ongoing inclusion within the pyOpenSci ecosystem.
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:107]:
```suggestion
with the community to identify reasonable next steps, such as assisting in finding a new maintainer. If a solution for ongoing package maintenance is not found, the package will be
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:111]:
```suggestion
working with the new maintainers to register the newly forked package within our ecosystem. The original package will be archived with a link to the new fork.
```

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:128]:
```suggestion
package from our ecosystem, we retain the right to offer the currently released version of the package in our
```

**Review by willingc — COMMENTED** [review]
I've made a few suggestions to help add clarity.

File: our-process/policies.md
Comment (pllim) [reply our-process/policies.md:94]: Do you have an idea what this icon would be? As @hamogu said, if a package is ultra-stable, 6-month period might be unfair for flaggin this package, especially if the icon can be something alarming like a ❌ .

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:27]: ok just coming back to this now. looking at this language we definitely would rather the review come to us prior to a journal article (altho not required) but most packages are already on pypi in our experience and we've been reviewing since 2019. we have one package that was only on bioconda (for very good reason because of dependencies). we could easily leave the text as is.

But the reality is that people often want to get the package name they selected. so they will push to pypi as a placeholder until the full package is done. should we adjust the. language to loosen up where it should be published first?

i also like published to a package repository such as PyPI or conda-forge as it leaves space for both / either. 

thoughts @hamogu @willingc ?

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:94]: got it. i'm thinking of a way to "watch" for repo activity. do you think if "update" meant either an issue comment or a commit / release 6 months would be reasonable. or do you still think we should wait a full year?

i was thinking something along the lines of 

1. 6 months - no activity - we just have a yellow flag or some shade of green that is not the "good to go" green. 
2. maybe we could then have a 12 months flag that turns red and we check in or maybe it's just orange at that point. 
there's are just markers for us to see what's going on with the package, not to actually archive the package from our ecosystem. thoughts?

Comment (lwasser) [reply our-process/policies.md:94]: ok i hear you all. we dont' want to alarm. i think (but let's talk about this). 

1. a 6 month flag - it can even be a small green flag that's a different shade of green that just says - hey this package hasn't been updated in 6 month (updated meaning no new commits, releases or issue activity??) i know issue activity could be from outside users so perhaps that is not a good marker
2. 12 month flag - maybe the status goes to yellow. that is when we'd directly reach out to maintainers. 

thoughts on something like that?

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:128]: @willingc question about this fix. i like the language but wonder - does this suggest that we are taking on maintenance of the package in this case? i guess i worry about that because we don't have the capacity to maintain new packages like rOpenSci does. But we might want to create an archive of the repository for safe keeping in the form of a fork, etc ? thoughts?

**Comment by lwasser** [comment]
ok everyone. i'm ready to tackle this one again. I've just integrated some of your changes into the archive policy!
I had a few questions around language. but i think we can merge this by the end of the week - (Friday dec 15) if we can just agree on the archive flag progression.

i'm proposing:

1. 6 months - no activity maybe a green flag appears. this just means we're watching the package but it's great so not a huge  issue just - hey activity is slow on the repo. this might be because there have been no commits / releases in 6 months (or issue activity??)
2. 12 months - after a year let's turn the flag from green to yellow. Still not red. At this point we'd reach out to the maintainer to determine next steps (if any are required). Perhaps we'll learn that there is no action needed and the flag goes back to green. 

the alternative is we just give maintainers a full year. (12 months) in terms of Ci buids that will be slightly easier to implement. thoughts?

File: our-process/policies.md
Comment (hamogu) [reply our-process/policies.md:94]: I think this suggestion is good. "shade of green" after 6 months of "nothing" (no issue, not code change, no release) and then "orange" after 12 month if there is no release (but even if there are issues and PRs. We follow-up manually and decide to give it another year or to archive). I think the decision to "archive" always warrants at least a cursory look by a human.

**Comment by hamogu** [comment]
Good point about implementation. 
I suggest to keep it simple - for us and for the package authors and only have one (12 month) cadence. Python is not evolving that fast any more. A package that has no activity after 6 month is not one that's so central to our ecosystem that we need to catch that immediately - and users are free to check number of stars, etc. if they want more information. This is just for an overview.

File: our-process/policies.md
Comment (willingc) [reply our-process/policies.md:27]: Published to a package repository makes sense to me. @lwasser

File: our-process/policies.md
Comment (willingc) [reply our-process/policies.md:128]: @lwasser I think it makes sense to be explicit about what we are doing. I don't think it is wise to imply that we will maintain any project. I was thinking along the lines of making a copy of the current release at the time of removal.

Comment (lwasser) [reply our-process/policies.md:128]: ahhh ok . How about language like:
***
> we retain the right to provide an archived copy of the package, in it's most recently release state the currently released version of the package in our ... 

:::{important}
pyOpenSci will never take on maintenance responsibilities of a community created package. But we will create archival copies of packages and support finding new maintainers for existing unmaintained packages as it makes sense. 
:::

***

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:94]: absolutely. i just thought as our community of packages grows (rOpenSci is up to 300) it will be hard to manually track that. so we can automate the initial flags and always manually override as it makes sense. i'll adjust the language just a bit locally and will push a change

Comment (lwasser) [reply our-process/policies.md:94]: new language - i'm pushing this up now!

> 
> pyOpenSci is building a system to track package metrics and activity, including issues, pull requests, and dates of
> the last release and last commit to the package repository. Activity is defined as a repository commit, pull request or release. 
> 
> We will 
> flag packages that haven't been updated on a 6-12 month time period based on activity as follows: 
> 
> - Packages with no activity for at least 6 months will receive a green flag on the website indicating that the package is still potentially maintained but no activity has been recorded in that time frame.  
> - Packages with no activity after 12 months will be flagged. At that time, pyOpenSci editorial team member will contact the package maintainers to evaluate the maintenance status of their package.

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:128]: i took this out of context - 

it now reads

> In the unlikely scenario that a contributor of a package requests removal of their
> package from our ecosystem, we retain the right offer the last / most recently released version of that package in our ecosystem for archival purposes only.

i forgot it said for archival purposes only and the way the gui presents things it only shows what has changed! this is fixed locally now

**Comment by lwasser** [comment]
> Good point about implementation. I suggest to keep it simple - for us and for the package authors and only have one (12 month) cadence. Python is not evolving that fast any more. A package that has no activity after 6 month is not one that's so central to our ecosystem that we need to catch that immediately - and users are free to check number of stars, etc. if they want more information. This is just for an overview.

ok - yes for implementation purposes it's much easier to track the last updated date. (i can add to our current system and write a bit of code to look for pr's, commits, releases ). then on the front end of the website, it can check the date, determine how long it's been since we've see "activity" and add or not add a flag to the package listing. This is a really easy check for me to build with our current infrastructure!

**Comment by lwasser** [comment]
language is now this - it's a cleaner policy AND easier to implement so i appreciate both

> 
> pyOpenSci is building a system to track package metrics and activity, including issues, pull requests, and dates of
> the last release and last commit to the package repository. Activity is defined as a repository commit, pull request or release.
> 
> We will flag packages that haven't been updated within a 1 year/ 12 month time period based on activity. Packages with no activity after 12 months will be flagged. At that time, pyOpenSci editorial team member will contact the package maintainers to evaluate the maintenance status of their package.

**Comment by lwasser** [comment]
i've also created a local task to do a bit more web dev when we add the astropy landing page to our site (after the APE is approved)

it will look something like this:

**BACKEND dev:**
* add a small module to update-web-meta with a last_activity_date. - this would parse a github repo and determine the latest date of a pr, commit or release
* the above would add a field to the packages.yml file last_activity_date
* 
**FRONT END DEV (what people see on the website)**

* parse the yaml file - find that field, and for each package, 
* if the date is older than 12 months, add a orange flag to the package card
* then simply add a key / explanation to the website that tells the user what the flag means.

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:33]: Maybe reframe slightly to:
By applying reviewer or editor recommendations, even previously published packages on PyPI, conda-forge, or in other venues can improve their user's experience with future version of the package.

File: our-process/policies.md
Comment (willingc) [inline our-process/policies.md:49]: Does this have meaning or is it stray text?

**Review by willingc — APPROVED** [review]
This is looking very good :D Thanks @lwasser

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:49]: oh @willingc i didn't understand this question but now i do

`(coi)=` is a [myst target](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#creating-explicit-targets). it allows me to create a short form link to the target location within the book rather than using a long header name. so i think this is a conflict of interest section target that i created.

**Comment by willingc** [comment]
Added https://github.com/pyOpenSci/software-peer-review/issues/268 to repo to track future revision steps.

Let's get this PR merged :smile:

File: our-process/policies.md
Comment (lwasser) [reply our-process/policies.md:33]: excellent. i suspect this might be more relevant to R / cran (our language started with language in the rOpenSci guide! i've reworded as follows:

> Applying reviewer or editor recommendations to your package can improve your user's experience with future versions of your package 
> even if your package is already published on `PyPI` or `conda-forge`.

**Comment by lwasser** [comment]
thank you so much @willingc i just pushed the last round of edits to this PR. and will merge now once all is "green" ✅  @hamogu @pllim thank you. both as well for the feedback and input. And please know as we implement the maintianed "Status" of a package we can always adjust our policies as it makes sense for all of our community partners and our maintainer community as well. but i'm very excited to know that the initial checks are something i know how to build based on our existing infrastructure!! so we really are not that far away from implementing this with our existing builds.

**Comment by lwasser** [comment]
All green !!! 🚢  :shipit: !!!

**Pull request merged**
